lib.h

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file uadi.h
+ * @brief Header file for Unified Abstract Dataproducer Interface (UaDI) library.
+ * @author Stephan Bökelmann
+ * @email sboekelmann@ep1.rub.de
+ *
+ * This file defines the API for interacting with various types of data producers.
+ * It includes functions for initializing the library, enumerating available data producers,
+ * claiming and releasing devices, managing data chunks, and waiting for data.
+ * Detailed error codes and data management policies are provided for robust integration.
+ */
+
+#ifndef UADI_H
+#define UADI_H
+
+#include <stddef.h>
+
+#ifdef _WIN32
+#define DLL_EXPORT __declspec(dllexport)
+#else
+#define DLL_EXPORT
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** 
+ * @brief Handle for the connection-id to the library instance.
+ *
+ * This is a void pointer to an instance that implements the DLL interface.
+ * The library is responsible for allocating and deallocating this handle.
+ */
+typedef void* uadi_lib_handle;
+
+/** 
+ * @brief Handle for a device instance.
+ *
+ * This pointer points to an instance that implements the interface of an abstract data producer.
+ * The library is responsible for managing the lifecycle of this handle.
+ * It gets created by claiming a device and gets destroyed after a device has been released by the consumer.
+ */
+typedef void* uadi_device_handle;
+
+/** 
+ * @brief Pointer to a chunk of memory.
+ *
+ * This is used to manage chunks of memory for data storage. Chunks ought to be contiguous
+ * in memory and at least 128KB in size. The client is responsible for allocating and
+ * deallocating these chunks.
+ */
+typedef unsigned char* uadi_chunk_ptr;
+
+/** 
+ * @brief Structure to receive data from the library.
+ *
+ * This structure is used when popping chunks of data from the library.
+ * It contains pointers to information and data packets. The format of data packets
+ * is an array of floats. Information packets are JSON strings.
+ * @see uadi_pop_chunk
+ */
+typedef struct {
+    uadi_chunk_ptr infopack_ptr;
+    uadi_chunk_ptr datapack_ptr;
+} uadi_receive_struct, *uadi_receive_struct_ptr;
+
+// Error codes
+#define UADI_SUCCESS 0
+#define UADI_ERROR -1
+#define UADI_BUFFER_TOO_SMALL -2
+#define UADI_INVALID_HANDLE -3
+#define UADI_NO_DATA -4
+#define UADI_OUT_OF_CHUNKS -5
+
+/**
+ * @brief Initialize the library and get a library handle.
+ * 
+ * @param buffer Pointer to a memory buffer for metadata. The buffer should be at least 128KB.
+ * @param buffer_length Length of the provided memory buffer.
+ * @return uadi_lib_handle A handle to the library instance. Returns nullptr if buffer is too small or other initialization errors occur.
+ */
+DLL_EXPORT uadi_lib_handle uadi_init(char* buffer, size_t buffer_length);
```

Comment:
use a second function to get the metadata. Would make library and user code more flexiable

Reply:
so `init()` gets no params at all?

Reply:
jop